### PR TITLE
feat: add `uploadcare` provider

### LINKED ISSUE #513: adding support for uploadcare provider
[Uploadcare](https://uploadcare.com/) comes pre-configured with Netlify CMS, popular for managing a Nuxt websites.

Trying to follow the doc to create a custom provider but it's a bit too obscure for my skill level.

Any help is welcome

Cheers,

### PR DESCRIPTION
**Pull request by luc122c**
This PR resolves #513 by implementing a new built-in provider for uploadcare.

 ### Operations 
 * [x] Image Compression
 * [x] Image Geometry
 * [ ] Image Overlays
 * [ ] Image Colours
 * [ ] Image Definition
 * [ ] Image Rotations
 
 ### Other stuff to think about later:
 * Signed URLs
 * File Groups
 * Custom CNAME: https://uploadcare.com/docs/delivery/cdn/#custom-cdn-cname

### CODE REVIEW

**Comment by luc122c** [comment]
Hi maintainers. I'm not sure if I've done this exactly right so I'd be gradeful for any tips/advice. I've tried to follow other providers as closely as I can. 👍

**Comment by danielroe** [comment]
Would you be able to merge the latest main and add snapshots/expected values? 🙏

**Comment by luc122c** [comment]
> Would you be able to merge the latest main and add snapshots/expected values? 🙏

I can certainly try. Please could you link to a good example of how you'd like it?

**Comment by danielroe** [comment]
Here's an example recent PR that included tests: https://github.com/nuxt/image/pull/774/files.

**Comment by luc122c** [comment]
Thanks! I'll give it a go :)

**Comment by luc122c** [comment]
@danielroe The tests are now passing, I hope I've done them correctly. 
This provider is in a suitable state to be used, I just haven't added some of the more advanced features yet. Would you like to merge it as-is and I can work on the other providers in a follow-up PR?

**Comment by danielroe** [comment]
Fantastic! Would you be able to add a page to the documentation? 🙏 

You could also include a note about what's usable, and maybe point people to make a PR if they would like.

File: docs/content/5.providers/uploadcare.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  border_radius: string | string[]
+  setfill: string // 3, 6 or 8 digit hex color
+  zoom_objects: string // 1 to 100
+```
+
+Please feel free to open a PR to improve support for additional operations.
+
+## Configuration
+
+> By default, all file URL's use the `ucarecdn.com` domain. By setting a custom CDN CNAME, the file URL's can use `cdn.mycompany.com` instead.
```
Comment (danielroe) [inline docs/content/5.providers/uploadcare.md:45]:
```suggestion
> By default, all file URLs use the `ucarecdn.com` domain. By setting a custom CDN CNAME, file URLs can use `cdn.mycompany.com` instead.
```

**Review by danielroe — APPROVED** [review]
Looks great to me! Thank you for your work on this ❤️

**Comment by luc122c** [comment]
No problem. Many thanks to you and the the other maintainers for a great module!

**Pull request merged**